bothan-core/src/store.rs

```diff
@@ -1,90 +1,228 @@
 use std::collections::{HashMap, HashSet};
+use std::path::Path;
+use std::sync::Arc;
 
-use tokio::sync::{Mutex, RwLock};
+use rust_rocksdb::{Options, DB};
+use tokio::sync::RwLock;
 
+pub use manager::ManagerStore;
+pub use worker::WorkerStore;
+
+use crate::registry::Registry;
+use crate::store::errors::Error;
+use crate::store::types::Key;
 use crate::types::AssetInfo;
-use crate::worker::AssetStatus;
 
-// TODO: Store should have namespace for different sources
-pub struct Store {
-    asset_store: RwLock<HashMap<String, AssetInfo>>,
-    query_ids: Mutex<HashSet<String>>,
+pub mod errors;
+mod manager;
+mod types;
+mod worker;
+
+#[derive(Clone)]
+pub struct SharedStore {
+    inner: Arc<RwLock<Inner>>,
 }
 
-impl Default for Store {
-    fn default() -> Self {
-        Store::new()
-    }
+struct Inner {
+    asset_store: HashMap<String, AssetInfo>,
```

Comment:
I don't think we need `asset_store`, `query_ids`, `active_signal_ids` as a separate hashmap now that we can have everything backed by `DB`? unless performance is a concern (which i don't think is the case since rocksdb is quite performant)